newlib/libc/sys/vita/pipe.c

```diff
@@ -75,4 +77,26 @@ int pipe(int pipefd[2])
     pipefd[1] = fd2;
 
     return 0;
+}
+
+
+int pipe2(int pipefd[2], int flags) {
+    if (flags & O_NONBLOCK) {
```

Comment:
don't we need to return -1 with `EFAULT`(or other error code) when both fds are not a socket in this case?

Reply:
fds are write-only, they don't need to be checked at entry...

Reply:
oh yeah. i got my mistake.